lib/test_queue/runner.rb

```diff
@@ -427,6 +429,18 @@ def distribute_queue
         else
           sock = @server.accept
           cmd = sock.gets.strip
+
+          token = cmd[TOKEN_REGEX, 1]
+          # If we have a slave from a different test run, respond with "WRONG RUN", and it will consider the test run done.
+          if token != @run_token
+            message = token.nil? ? "Worker sent no token to master" : "Worker from run #{run_token} connected to master"
+            STDERR.puts "*** #{message} for run #{@run_token}; ignoring."
+            sock.write("WRONG RUN\n")
+            next
+          end
+
+          cmd = cmd.gsub(TOKEN_REGEX, "")
```

Comment:
Did you see races happening? I'd be a bit surprised since each connection to the master is only used once then thrown away. I'm no networking expert but I didn't think distinct TCP connections could interfere with each other.